### PR TITLE
allow giving momentum to specific team without joining

### PR DESCRIPTION
**Pull request by bmorel**
This is not the cleaner implementation, but it works. I think it's not *much* of a problem if a cheat command is not cleanly implemented, and overall this whole file should probably be done the same way sg_cmds.cpp is done, as said in #538.
Still, this patch is useful as it reduces a lot the pain when doing tests (checking if events bound to momentum or if a specific change in AI works correctly, notably, I don't really see other reasons to use that).

### CODE REVIEW

**Comment by Gireen** [comment]
what happens if the `give` other commands are used as spectator?

**Comment by bmorel** [comment]
Good question. I just tried a `give all`, and it simply behave as I would expect: makes credits at their maximum value.

File: src/sgame/sg_cmds.cpp [hunk 1]
```diff
@@ -593,12 +593,13 @@ void Cmd_Give_f( gentity_t *ent )
 	char     *name;
 	bool give_all = false;
 	float    amount;
+	team_t team = static_cast<team_t>( ent->client->pers.team );
```
Comment (necessarily-equal) [inline src/sgame/sg_cmds.cpp:596]:
```suggestion
	team_t team = G_Team(ent);
```

**Review by necessarily-equal — COMMENTED** [review]
I would replace all the references to `TEAM_NONE` using `BG_IsPlayableTeam`, which is coherent with `BG_PlayableTeamFromString`.

The error reporting is bad, but to be honest, that's a problem of this function in the first place, and it can be done in another PR as this would require a refactor.

**Comment by bmorel** [comment]
This function, and others in that file, do not just have bad error reporting actually. This stuff likely requires real work to get cleaned.

**Comment by bmorel** [comment]
Are there still problems with this PR?

**Comment by bmorel** [comment]
Any reason to not merge it?

**Comment by necessarily-equal** [comment]
There is still this weird thing where once you use `G_IsPlayableTeam` 
and once `team != TEAM_NONE`. You can merge it anyway though

**Comment by bmorel** [comment]
Fixed.

**Pull request merged**